### ISSUE TITLE
env/strenv with unresolved variable

### ISSUE THREAD

**Issue by hobti01** [mikefarah/yq#651]
**Please describe your feature request.**
Somehow clarify when variables are available and/or the error when a variable has no value.
* I suppose the variable availability is shell specific?

Note:
- how to questions should be posted in the discussion board and not raised as an issue.
- V3 will no longer have any enhancements.

**Describe the solution you'd like**
With data1.yml:

```yaml
.foo: dot
foo: nodot
```

command:

```bash
$ var=foo yq eval '.[env(var)]' data1.yaml 
nodot
$ var=foo; yq eval '.[env(var)]' data1.yaml 
Error: EOF
# alternative:
$ var=foo
$ yq eval '.[env(var)]' data1.yaml 
Error: EOF
# but
$ export var=foo
$ yq eval '.[env(var)]' data1.yaml 
nodot
```

it could output

```bash
Error: value not provided to `env()`
```

**Describe alternatives you've considered**
Requiring an export variables reduces the usefulness of `env` and `strenv`

**Additional context**
Thanks for these new operators, they are very promising!

**Comment by mikefarah** [comment]
Nice - will add it to the next release

**Comment by mikefarah** [comment]
Fixed in 4.3.2
